server/src/main/java/org/apache/iotdb/db/query/aggregation/AggregateResult.java

```diff
@@ -34,12 +34,19 @@
 import java.io.OutputStream;
 import java.nio.ByteBuffer;
 
-public abstract class AggregateResult {
+public abstract class AggregateResult implements Cloneable {
 
   public static final int TIME_LENGTH_FOR_FIRST_VALUE = 100;
   private final AggregationType aggregationType;
   protected TSDataType resultDataType;
 
+  /**
+   * For [COUNT, AVG, SUM], it is the start time of the aggregation window. For [MAX_VALUE,
+   * MIN_VALUE, EXTREME, FIRST_VALUE, LAST_VALUE], it is the timestamp of the current value. For
+   * [MAX_TIME, MIN_TIME], it is always null.
+   */
+  protected long timestamp;
```

Comment:
remember to check whether it's ok to use in case of left open interval.

Reply:
I has checked ~ no problem